### PR TITLE
feat(cb2-6886): prevent test creation on skeleton tech records

### PR DESCRIPTION
**Pull request by tahayazici**
## Ensure contingency tests can't be captured on skeleton Tech Records in VTM

- Add a completeness tag to the tech record title.
- Remove the redundant create test record button component.
- Display a pop-up when the user tries to create a test on a skeleton tech record.

[CB2-6869](https://dvsa.atlassian.net/browse/CB2-6869)
[CB2-6886](https://dvsa.atlassian.net/browse/CB2-6886)

### CODE REVIEW

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts [hunk 1]
```diff
@@ -106,7 +108,32 @@ export class VehicleTechnicalRecordComponent implements OnInit, AfterViewInit {
...
+      default:
+        return 'Unknown Vehicle Type';
+    }
+  }
+
+  createTest(isComplete?: string): void {
+    if (isComplete) {
+      this.router.navigate(['test-records/create-test/type'], { relativeTo: this.route });
+    } else {
+      confirm(
```
Comment (Ian-Hodges) [inline src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:128]: Would alert() be better? confirm() suggests they have an option to continue or cancel.

Comment (tahayazici) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts]: Suggestion applied.

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts [hunk 2]
```diff
@@ -106,7 +108,32 @@ export class VehicleTechnicalRecordComponent implements OnInit, AfterViewInit {
...
+      default:
+        return 'Unknown Vehicle Type';
+    }
+  }
+
+  createTest(isComplete?: string): void {
+    if (isComplete) {
+      this.router.navigate(['test-records/create-test/type'], { relativeTo: this.route });
+    } else {
+      alert(
```
Comment (aaron-downing) [inline src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:128]: Should it be alert or should we just add a red banner box at the top of the page?

Comment (aaron-downing) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:128]: A bit like when you try to submit amends on a tech record when you're missing fields?

Comment (tahayazici) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:128]: It should be an alert, that's what they want. Check the ticket.

Comment (aaron-downing) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:128]: But that SD is from an app, might be worth just asking Matt

Comment (tahayazici) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:128]: Just got confirmation from Matt. We're going ahead with this.

**Review by Ian-Hodges — APPROVED** [review]

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts [hunk 3]
```diff
@@ -106,7 +108,32 @@ export class VehicleTechnicalRecordComponent implements OnInit, AfterViewInit {
...
+      case VehicleTypes.HGV:
+        return techRecord.make && techRecord.model ? `${techRecord.make}-${techRecord.model}` : '';
+      default:
+        return 'Unknown Vehicle Type';
+    }
+  }
+
+  createTest(isComplete?: string): void {
+    if (isComplete) {
+      this.router.navigate(['test-records/create-test/type'], { relativeTo: this.route });
```
Comment (aaron-downing) [inline src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:126]: very minor, but you could use `return this.router.navigate(['test-records/create-test/type'], { relativeTo: this.route });` and not need the else statement

Comment (tahayazici) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:126]: That is semantically wrong or confusing as this method's return type is `void`. That change would only get rid of the `} else {` line, so I'll leave it as is for now.

**Review by leifkemp-bjss — APPROVED** [review]

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.html [hunk 4]
```diff
@@ -1,31 +1,26 @@
...
+        <app-tech-record-history [currentRecord]="currentTechRecord" [vehicleTechRecord]="vehicleTechRecord"></app-tech-record-history>
 
-        <app-test-record-summary *appRoleRequired="roles.TestResultView" [testRecords]="(records$ | async) || []"> </app-test-record-summary>
+        <app-test-record-summary *appRoleRequired="roles.TestResultView" [testRecords]="(records$ | async) || []"></app-test-record-summary>
 
         <ng-container *ngIf="!isArchived && !isEditing">
-          <app-test-record-create-button *appRoleRequired="[roles.TestResultCreateContingency, roles.TestResultCreateDeskAssesment]">
-          </app-test-record-create-button>
+          <div class="govuk-button-group" *appRoleRequired="[roles.TestResultCreateContingency, roles.TestResultCreateDeskAssesment]">
+            <a id="create-test" class="govuk-button" (click)="createTest(currentTechRecord?.recordCompleteness)">Create a test for this vehicle</a>
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.html:22]: This should be a button, or it should have an `href` or `routerLink` attribute.

File: src/app/features/tech-record/components/tech-record-title/tech-record-title.component.ts [hunk 5]
```diff
@@ -41,6 +42,14 @@ export class TechRecordTitleComponent implements OnInit {
     return Roles;
   }
 
+  getCompletenessColor(completeness?: string): 'green' | 'red' {
```
Comment (tom-bjss) [inline src/app/features/tech-record/components/tech-record-title/tech-record-title.component.ts:45]: We should add this to our own technical debt register, Im sure we do something similar for Test results, good opportunity to refactor in the future 😄

**Review by tom-bjss — APPROVED** [review]

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.html (hunk 4 above)
Comment (tahayazici) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.html:22]: The hover and cursor effect is already handled by the gov styling, so it is not necessary.

File: src/app/features/tech-record/components/tech-record-title/tech-record-title.component.ts (hunk 5 above)
Comment (tahayazici) [reply src/app/features/tech-record/components/tech-record-title/tech-record-title.component.ts:45]: I do not really get what you mean, but sure, yeah 😊

**Pull request merged**